services/api/service.go

```diff
@@ -421,12 +422,25 @@ func (api *RelayAPI) StartServer() (err error) {
 		}
 	}
 
-	if !foundCapellaEpoch || !foundDenebEpoch {
-		return ErrMissingForkVersions
+	// feature flags
+	ffCheckDenebForkSchedule := false // compatibility with older CL clients missing deneb fork schedule
+	if os.Getenv("CHECK_DENEB_FORK_SCHEDULE") != "" {
+		log.Warn("env: CHECK_DENEB_FORK_SCHEDULE: checking validation for deneb fork schedule")
+		ffCheckDenebForkSchedule = true
+	}
+
+	if ffCheckDenebForkSchedule {
+		if !foundCapellaEpoch || !foundDenebEpoch {
```

Comment:
do we even need these checks? in which case do we need to return `ErrMissingForkVersions`?

Reply:
we return `ErrMissingForkVersions` when:
1. Flag check is on: we require the beacon node to have both capella and Deneb fork schedule returned which will assign the two variables = true
2. Flag check is off: we only require the beacon node to have the capella fork schedule and don't care about whether the Deneb epoch was assigned.

This check allows relay to error early if the beacon node is outdated and did not return the fork schedules we expect. However this breaks on mainnet as the Deneb fork schedule is not set yet.

Reply:
how about we just remove this check all together? we might not need to babysit the relay operators, and this logic is already a bit confusing and prone to errors 🤔 

what if we remove both `ErrMissingForkVersions` and `REQUIRE_DENEB_FORK_SCHEDULE`? this would make things around here considerably simpler